src/api/common/HttpClient.ts

```diff
@@ -0,0 +1,73 @@
+import log from 'loglevel';
+import type {
+    HubInfo,
+    RelayEstimation,
+    RelayOrDeployRequest
+} from 'src/common/relay.types';
+import type HttpWrapper from './HttpWrapper';
+
+const PATHS = {
+  GET_INFO: '/getaddr',
+  POST_REALY_REQEUST: '/relay',
+  POST_ESTIMATE: '/estimate'
+} as const;
+const VERIFIER_SUFFIX = '?verifier=';
+
+export default class HttpClient {
+  private readonly _httpWrapper: HttpWrapper;
+
+  constructor(httpWrapper: HttpWrapper) {
+    this._httpWrapper = httpWrapper;
+  }
+
+  async getHubInfo(relayUrl: string, verifier = ''): Promise<HubInfo> {
+    const verifierSuffix = verifier && `${VERIFIER_SUFFIX}${verifier}`;
+    const hubInfo: HubInfo = await this._httpWrapper.sendPromise(
+      relayUrl + PATHS.GET_INFO + verifierSuffix
+    );
+    if (!hubInfo) {
```

Comment:
Why don't we check for the error here?

Reply:
Bit worried about [this error handling](https://github.com/rsksmart/rif-relay-server/blob/master/src/HttpServer.ts#L181) in the server, which also points out the uselessness of this check here on L28. We should probably aim at changing that to something more sensible.